### ISSUE TITLE
Bug suspected in Microsoft API when enabling storage threat protection - error parsing URI

### ISSUE THREAD

**Issue by peteneville** [terraform-providers/terraform-provider-azurerm#10179]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

Terraform v0.12.30
provider.azurerm v2.42.0
### Terraform Configuration Files

```hcl
resource "azurerm_advanced_threat_protection" "business" {
  target_resource_id = azurerm_storage_account.business.id
  enabled            = true
}
```

### Expected Behaviour

Advanced Threat Protection should successfully be enabled for the given storage account

### Actual Behaviour
Terraform errored with:

Error: [ERROR] Unable to parse Advanced Threat Protection Set ID "subscriptions/ece6...c45/resourceGroups/test_xyz_shared/providers/Microsoft.Storage/storageAccounts/xyzbusiness/providers/Microsoft.Security/advancedThreatProtectionSettings/current": Cannot parse Azure ID: parse "subscriptions/ece6...c45/resourceGroups/test_xyz_shared/providers/Microsoft.Storage/storageAccounts/xyzbusiness/providers/Microsoft.Security/advancedThreatProtectionSettings/current": invalid URI for request

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

State file showed the resource as tainted and looking closely we can see the leading '/' is missing on the id (ignore the ...):

      "mode": "managed",
      "type": "azurerm_advanced_threat_protection",
      "name": "business",
      "provider": "provider.azurerm",
      "instances": [
        {
          "status": "tainted",
        ...
            "id": "subscriptions/ece6...c45/resourceGroups/..../providers/Microsoft.Storage/storageAccounts/..../providers/Microsoft.Security/advancedThreatProtectionSettings/current",
            "target_resource_id": "/subscriptions/ece6...45/resourceGroups/..../providers/Microsoft.Storage/storageAccounts/...",
...
    },

Updating the state file by adding the leading '/', re-running the apply and it works.

Noticed that the unit test in resourceid_test.go checks for missing leading '/' so can only assume that the Microsoft API is broken.
Whilst this may not be a terraform provider bug, the community here should be aware and also might be in a position to help.

**Comment by sbl-matter** [comment]
Thanks, I have the same problem with different versions of azurerm provider and terraform (2.14.0 and 0.12.29).

Is it safe to manually update the state file ? Or should we wait for a fix ?

**Comment by csdaraujo** [comment]
Also happens with the lastest provider: 2.42.0.

**Comment by csdaraujo** [comment]
Also happens when using TF 13 (13.5).

**Comment by peteneville** [comment]
> Thanks, I have the same problem with different versions of azurerm provider and terraform (2.14.0 and 0.12.29).
> 
> Is it safe to manually update the state file ? Or should we wait for a fix ?

Hi, For a production system, I'd wait for a fix unless it's preventing an urgent change. Hopefully you can test on a non-production system / take a backup of the state file etc if you needed to.

**Comment by danielbmeireles** [comment]
We were able to workaround this issue editing the state directly:

FROM:

```
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "enabled": true,
            "id": "subscriptions/4f6b07a7-ddf5-4f6c-a0f1-1b23cfea2ac3/resourceGroups/rg-dev-global-services/providers/Microsoft.Storage/storageAccounts/stdevflink/providers/Microsoft.Security/advancedThreatProtectionSettings/current",
            "target_resource_id": "/subscriptions/4f6b07a7-ddf5-4f6c-a0f1-1b23cfea2ac3/resourceGroups/rg-dev-global-services/providers/Microsoft.Storage/storageAccounts/stdevflink",
            "timeouts": null
          },
```

TO:

```
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "enabled": true,
            "id": "/subscriptions/4f6b07a7-ddf5-4f6c-a0f1-1b23cfea2ac3/resourceGroups/rg-dev-global-services/providers/Microsoft.Storage/storageAccounts/stdevflink/providers/Microsoft.Security/advancedThreatProtectionSettings/current",
            "target_resource_id": "/subscriptions/4f6b07a7-ddf5-4f6c-a0f1-1b23cfea2ac3/resourceGroups/rg-dev-global-services/providers/Microsoft.Storage/storageAccounts/stdevflink",
            "timeouts": null
          },
```

**Comment by danielbmeireles** [comment]
We were able to workaround this issue editing the state directly:

FROM:

```
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "enabled": true,
            "id": "subscriptions/SUBSCRIPTION_ID/resourceGroups/RG_NAME/providers/Microsoft.Storage/storageAccounts/STRACC_NAME/providers/Microsoft.Security/advancedThreatProtectionSettings/current",
            "target_resource_id": "/subscriptions/SUBSCRIPTION_ID/resourceGroups/RG_NAME/providers/Microsoft.Storage/storageAccounts/STRACC_NAME",
            "timeouts": null
          },
```

TO:

```
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "enabled": true,
            "id": "/subscriptions/SUBSCRIPTION_ID/resourceGroups/RG_NAME/providers/Microsoft.Storage/storageAccounts/STRACC_NAME/providers/Microsoft.Security/advancedThreatProtectionSettings/current",
            "target_resource_id": "/subscriptions/SUBSCRIPTION_ID/resourceGroups/RG_NAME/providers/Microsoft.Storage/storageAccounts/STRACC_NAME",
            "timeouts": null
          },
```

**Comment by jamescross91** [comment]
For us (Terraform 13.5, AzureRM 2.36.0) this was solved by upgrading to AzureRM 2.40.0
